test/e2e/experimental_features.go

```diff
@@ -61,9 +61,11 @@ func ExperimentalFeaturesSpec(ctx context.Context, inputGetter func() Experiment
 		Expect(input.ClusterctlConfigPath).To(BeAnExistingFile(), "Invalid argument. input.ClusterctlConfigPath must be an existing file when calling %s spec", specName)
 		Expect(input.BootstrapClusterProxy).ToNot(BeNil(), "Invalid argument. input.BootstrapClusterProxy can't be nil when calling %s spec", specName)
 		Expect(os.MkdirAll(input.ArtifactFolder, 0755)).To(Succeed(), "Invalid argument. input.ArtifactFolder can't be created for %s spec", specName)
-
 		Expect(input.E2EConfig.Variables).To(HaveKey(KubernetesVersion))
+
+		// Read CNI file and set CNI_RESOURCES environmental variable
 		Expect(input.E2EConfig.Variables).To(HaveKey(CNIPath))
+		setCNIResources(input.E2EConfig.GetVariable(CNIPath))
```

Comment:
Given that we are testing ClusterResourceSet in all tests, could we drop entirely the `ExperimentalFeatures` E2E test?

Reply:
We are not testing ClusterResourceSetBinding during CNI install. If we add this test in ApplyClusterTemplateAndWait, we can drop it. But I prefer not to, as with `Sync` mode, we will need more things to test.

Reply:
I agree that we are losing an E2E test on the internal details of ClusterResourceSet (the binding), but this is something that is already tested by the unit tests.

IMO, what is important for E2E is to test that ClusterResourceSet does what is expected (in this case it installs the CNI), given that we are doing this for all the E2E tests I think that we have good coverage of this feature.
Also, E2E test is a trade-off between what we want to test and available resources/test duration, and, given the above consideration, I think it is fine to drop this test.

Reply:
Makes sense, I dropped the tests.